frontend/server/src/DAO/Submissions.php

```diff
@@ -313,4 +313,34 @@ public static function getLatestSubmissions(
             'submissions' => $submissions,
         ];
     }
+
+    /** @return list<array{time: \OmegaUp\Timestamp}> */
+    public static function getLastSubmissionByProblem(
+        int $problemId,
+        ?int $identityId
+    ): array {
+        $params = [$problemId];
+        $sql = '
+            SELECT
+                s.time
+            FROM
+                Submissions s
+            WHERE
+                s.problem_id = ?
+        ';
+
+        if (!is_null($identityId)) {
+            $params[] = $identityId;
+            $sql .= ' AND s.identity_id = ?';
+        }
+
+        $sql .= '
+            ORDER BY
+                s.time DESC
+            LIMIT 1
+        ';
```

Comment:
En que caso `$identityId` podría ser null? Para que un usuario pueda realizar un envío debe estar logueado

Así que no debería ser necesaria la condición:
```suggestion
        $sql = '
            SELECT
                s.time
            FROM
                Submissions s
            WHERE
                s.problem_id = ?
                AND s.identity_id = ?
            ORDER BY
                s.time DESC
            LIMIT 1
        ';
```

Reply:
Oh cierto, en realidad estaba pensando en hacer un método genérico, pero creo que no tiene sentido, cambiaré eso entonces.